w2w/w2w.py

```diff
@@ -174,32 +186,103 @@ def main(argv=None):
     print("********* All done ***********")
 
 
-def check_lcz_wrf_extent(info: Dict[str, str]) -> None:
 
-    # Read the data
-    lcz = rasterio.open(info['src_file'])
-    wrf = xr.open_dataset(info['dst_file'])
+# Overall function that tests the LCZ integrity, including:
+# - check LCZ classes, change if needed
+# - check projection system, change if needed
+# - check extent
+
+def _replace_lcz_number(lcz, lcz_to_change):
+
+    lcz_expected = np.array(
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+         11, 12, 13, 14, 15, 16, 17]
+    )
+
+    lcz_arr = lcz.data.flatten().astype('int')
+    df = pd.Series(lcz_arr, dtype=lcz_expected.dtype)
+
+    d = dict(zip(lcz_to_change, lcz_expected))
+
+    lcz_new = lcz.copy()
+    lcz_new.data = df.map(d).values\
+        .reshape(lcz.shape)\
+        .astype('int8')
+
+    return lcz_new
+
+def _check_lcz_wrf_extent(lcz, wrf) -> str:
 
     # Get bounding box coordinates
-    lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax = lcz.bounds
+    lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax = lcz.rio.bounds()
     wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax = \
         float(wrf.XLONG_M.min()), float(wrf.XLAT_M.min()), \
         float(wrf.XLONG_M.max()), float(wrf.XLAT_M.max())
 
-    # Evaluate and throw error if wrf not within LCZ domain
+    # Evaluate and throw error if WRF not within LCZ domain
     if not (wrf_xmin > lcz_xmin ) & (wrf_xmax < lcz_xmax ) & \
            (wrf_ymin > lcz_ymin) & (wrf_ymax < lcz_ymax):
 
-        print("ERROR: LCZ domain should be larger than WRF domain "
-              "in all directions.")
-        print(f"LCZ bounds  (xmin, ymin, xmax, ymax): "
-              f"{lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax}")
-        print(f"WRF bounds  (xmin, ymin, xmax, ymax): "
-              f"{wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax}")
+        message = "ERROR: LCZ domain should be larger than WRF domain " \
+                  "in all directions.\n" \
+                  "LCZ bounds (xmin, ymin, xmax, ymax): " \
+                 f"{lcz_xmin, lcz_ymin, lcz_xmax, lcz_ymax}\n" \
+                  "WRF bounds (xmin, ymin, xmax, ymax): "\
+                 f"{wrf_xmin, wrf_ymin, wrf_xmax, wrf_ymax}"
+    else:
+        message = "OK - LCZ domain is covering WRF domain"
+
+    return message
```

Comment:
I think we can already raise a `SystemExit(message)` here if the error comes up, no need to return the message first. If no error comes up, just print the message and return nothing (`None`)  or just `True`.

Reply:
Adjusted